perma_web/perma/celery_tasks.py

```diff
@@ -1577,8 +1593,41 @@ def warn_expiring_sponsored_users(warning_days=None):
                     "expiration_date": expiration_date + timedelta(days=1)
                 }
                 try:
-                    email_expiring_sponsored_user(sponsorship.user, context)
+                    email_expiring_user(sponsorship.user, "email/sponsored_user_expiry_notification.txt", context)
                 except Exception as e:
                     logger.error(f"Error emailing user {sponsorship.id}: {e}")
                 break
 
+
+@shared_task
+def warn_expiring_organization_users(warning_days=None):
+    """
+    Notifies users whose affiliation with an organization is expiring
+    """
+    if not warning_days:
+        warning_days = [7, 15, 30]
+
+    todays_date = timezone.now().date()
+    max_notification_date = todays_date + timedelta(days=max(warning_days))
+    expired_organization_users = UserOrganizationAffiliation.objects.filter(expires_at__lt=max_notification_date)
+
+    if not expired_organization_users:
+        return
+
+    for affiliation in expired_organization_users:
+        expiration_date = affiliation.expires_at.date()
+        for day in warning_days:
+            warning_date = todays_date + timedelta(days=day)
+            if expiration_date == warning_date:
+                context = {
+                    "organization_name": affiliation.organization.name,
+                    "registrar_name": affiliation.organization.registrar.name,
+                    "registrar_email": affiliation.organization.registrar.email,
+                    "expiration_days": day,
+                    "expiration_date": expiration_date + timedelta(days=1)
+                }
+                try:
+                    email_expiring_user(affiliation.user, "email/organization_user_expiry_notification.txt", context)
+                except Exception as e:
+                    logger.error(f"Error emailing user {affiliation.user_id}: {e}")
```

Comment:
We might consider using `logger.exception` so the full traceback gets included:
```suggestion
                except Exception:
                    logger.exception(f"Error emailing user {affiliation.user_id}")
```
The error might be obvious in most cases, though — I'll leave this up to you.